pandas/core/dtypes/cast.py

```diff
@@ -2074,7 +2083,7 @@ def maybe_cast_to_integer_array(
     if is_object_dtype(arr.dtype):
         raise ValueError("Trying to coerce float values to integers")
 
-    if casted.dtype < arr.dtype:
+    if casted.dtype < arr.dtype or lib.infer_dtype(casted) < lib.infer_dtype(arr):
```

Comment:
`lib.infer_dtype(casted) < lib.infer_dtype(arr)` im not sure what this is supposed to mean